File: golang/task1/14mod1.go
package main

import (
	"bufio"   // для удобного чтения ввода с клавиатуры (ввод построчно)
	"fmt"     // для вывода на экран и форматирования строк
	"os"      // для работы с операционной системой (доступ к stdin)
	"strconv" // для преобразования строки в число
	"strings" // для обработки строк (удаление пробелов, работа с подстроками)
)

func towers(str string, x int) bool {
	n := len(str)
	cover := make([]bool, n) //Создаем сновной массив покрытия

	for i := 0; i < n; i++ {
		if str[i] == '1' {
			cover[i] = true
		} else {
			cover[i] = false
		}
	}

	for i := 0; i < n; i++ {
		if cover[i] {
			for k := 0; k < x; k++ {
				if i+x < n {
					cover[i+k] = true
				}
				if i-x >= 0 {
					cover[i-k] = true
				}
			}
		}
	}
	for i := 0; i < n; i++ {
		if !cover[i] {
			return false
		}
	}
	return true
}

func main() {
	scanner := bufio.NewScanner(os.Stdin) // Ининциализируем пользовательский ввод
	fmt.Println("Введите вышки: ")
	scanner.Scan()
	s := strings.TrimSpace(scanner.Text())

	fmt.Println("Введите показатель X: ")
	scanner.Scan()
	xStr := strings.TrimSpace(scanner.Text()) // X считывается как строка
	x, err := strconv.Atoi(xStr)              // Строка преобразуется в целое число
	if err != nil {                           // Проверка на правильность строки (Х - целое число)
		fmt.Println("Строка должна быть целым числом")
		return
	}

	if towers(s, x) {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}

File: php/task1/14mod1.php
<?php
function towers($str, $x) { // Функция проверки покрытия
    $n = strlen($str); // Длина строки
    $cover = array(); // Массив покрытия

    for ($i = 0; $i < $n; $i++) {
        if ($str[$i] == '1') {
            $cover[$i] = true;
        } else {
            $cover[$i] = false;
        }
    }

    for ($i = 0; $i < $n; $i++) { // Обновление покрытия
        if ($cover[$i] == true) {
            for ($k = 0; $k <= $x; $k++) {
                if (isset($cover[$i + $k])) $cover[$i + $k] = true; // Покрытие вправо
                if (isset($cover[$i - $k])) $cover[$i - $k] = true; // Покрытие влево
            }
        }
    }

    for ($i = 0; $i < $n; $i++) { // Проверка покрытия
        if ($cover[$i]) continue;
        else return false;
    }

    return true;
}

// Main
echo "Введите вышки:\n";
$s = trim(fgets(STDIN)); // Считываем строку 
echo "Введите показатель x:\n";
$x = intval(trim(fgets(STDIN))); // Считываем число x

if (towers($s, $x)) {
    echo "Yes\n";
} else {
    echo "No\n";
}
?>
